libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -525,10 +586,29 @@ void AP_Baro::update(void)
         sensors[i].healthy = (now - sensors[i].last_update_ms < 500) && !is_zero(sensors[i].pressure);
     }
 
+    // find current GPS altitude
+    float gps_alt = 0.0f;
+    bool process_gps_alt = false;
+    if (!is_zero(_gps_adj_step) && _gps_calibrated) {
+        float gps_vacc;
+        if (_gps->vertical_accuracy(gps_vacc)) {
```

Comment:
_gps-> is only used in two functions, perhaps instead of passing in the reference (causing a dependancy) we can just pass in the 3 variables and/or a reference in the cal and update()?